### PR TITLE
Make it easy to test locally

### PR DESCRIPTION
**Pull request by stsewd**
Additionally, disabling the CORS check should be done.
I'm using https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere/ for that.

### CODE REVIEW

**Comment by stsewd** [comment]
Looks like sphinx 3.5 broke the extension :) taking a look

**Comment by stsewd** [comment]
Looks like they updated underscore.js, I'll check the incompatible changes, also see if we can drop that dep.

File: docs/development.rst [hunk 1]
```diff
@@ -23,6 +23,17 @@ Generate minified JS and CSS files via ``Gulp``:
...
 available at :doc:`Testing page <testing>`.
 
+Local testing
+-------------
+
+This extension works when is hosted on Read the Docs,
+to make it work locally a custom ``READTHEDOCS_DATA`` js variable is provided to send the search requests to
+https://readthedocs-sphinx-search.readthedocs.io/_/api/v2/search/.
+Due to our CORS settings your browser will block the requests,
+so you'll need to install a plugin like `CORS everywhere`_ to make it work.
```
Comment (ericholscher) [inline docs/development.rst:33]: Can we use jsonp in this case to work around this issue?

Comment (stsewd) [reply docs/development.rst:33]: Just tested it, our backend needs to respond with `application/jsonp` as well, thing we don't support in our api.

Comment (ericholscher) [reply docs/development.rst:33]: Sure, but most of our API endpoints support jsonp, I don't think it would be hard to support here, and much less hacky :)

Comment (ericholscher) [reply docs/development.rst:33]: Isn't it just adding this: 

https://github.com/readthedocs/readthedocs.org/blob/8e78d680d02aeba12644796b979ef62459f64932/readthedocs/api/v2/views/footer_views.py#L91

Comment (stsewd) [reply docs/development.rst:33]: yeah, we would need to add that, but I'm not really in favor of solving our cors problems with jsonp, maybe we should just add the cors header to the api to allow all domains instead.

Comment (ericholscher) [reply docs/development.rst:33]: That is probably a better solution 👍

Comment (stsewd) [reply docs/development.rst]: We have CORS header now! I have removed this section now.

File: sphinx_search/static/js/rtd_sphinx_search.js [hunk 2]
```diff
@@ -697,7 +697,7 @@ window.addEventListener("DOMContentLoaded", () => {
         const project = READTHEDOCS_DATA.project;
         const version = READTHEDOCS_DATA.version;
         const language = READTHEDOCS_DATA.language || "en";
-        const api_host = '/_';
+        const api_host = READTHEDOCS_DATA.proxied_api_host || '/_';
```
Comment (ericholscher) [inline sphinx_search/static/js/rtd_sphinx_search.js:700]: I'd love to standardize this a bit more in some kind of JS client, at some point in the future.

**Review by ericholscher — APPROVED** [review]
Looks like a good change 👍

File: docs/development.rst [hunk 3]
```diff
@@ -23,6 +23,26 @@ Generate minified JS and CSS files via ``Gulp``:
...
+Local testing
+-------------
+
+You can test this extension from the docs folder:
+
+.. prompt:: bash
+
+   cd docs
+   pip install sphinx-autobuild
+   pip install -r requirement.txt
```
Comment (nienn) [inline docs/development.rst:35]: Isn't it the name of this file **requirements.txt** (plural)?

Comment (astrojuanlu) [reply docs/development.rst:35]: Good catch @nienn!

**Pull request merged**